File: example_apps/HanksTodo/src/sapphire/appexamples/hankstodo/app/TodoList.java
package sapphire.appexamples.hankstodo.app;

import java.io.Serializable;
import java.util.ArrayList;

import sapphire.app.*;
import sapphire.policy.cache.CacheLeasePolicy;

public class TodoList implements SapphireObject<CacheLeasePolicy>{ 
	ArrayList<Object> toDos = new ArrayList<Object>();
	String name = "Hanks todo";

	public TodoList(String name) {
		toDos = new ArrayList<Object>();
		this.name = name;
	}

	public String addToDo(String todo) {
		System.out.println("Inside todo: " + todo);
		toDos.add(todo);
		return "OK!";
	}

	public void completeToDo(String todo) {
	}

	public ArrayList<Object> getHighPriority() {
		return new ArrayList<Object>();
	}
}

File: multilanguage/demo_grpc/go/hankstodo/app/TodoList.go
package main

import (
	"fmt"
)

type TodoList struct {
	ToDos []string
	Name  string
}

func (t *TodoList) AddToDo(todo string) string {
	fmt.Println("Inside of AddToDo")
	fmt.Println(todo)
	t.ToDos = append(t.ToDos, todo)
	return "Ok!"
}
